scanner/cmd/scanner/main.go

```diff
@@ -238,7 +244,47 @@ func getServices(b *Backends) []grpc.APIService {
 	return srvs
 }
 
-// Close closes all the backends used by the scanner.
+// HealthCheck returns true if all configured backends are healthy and ready.
+func (b *Backends) HealthCheck(ctx context.Context) bool {
+	var checkList []func(context.Context) error
+	if b.Indexer != nil {
+		checkList = append(checkList, b.Indexer.Ready)
+	}
+	if b.Matcher != nil {
+		checkList = append(checkList, b.Matcher.Ready)
+	}
+	for _, check := range checkList {
+		if err := check(ctx); err != nil {
+			zlog.Error(ctx).Err(err).Msg("scanner is not ready")
+			return false
+		}
+	}
+	return true
+}
+
+// HealthRoutes returns HTTP routes for health checking the configured backends.
+func (b *Backends) HealthRoutes() (r []routes.CustomRoute) {
+	for n, h := range map[string]http.HandlerFunc{
+		"/health/liveness": func(w http.ResponseWriter, r *http.Request) {
+			// If we can reply, we are alive.
+			w.WriteHeader(http.StatusOK)
+		},
+		"/health/readiness": func(w http.ResponseWriter, r *http.Request) {
+			if !b.HealthCheck(r.Context()) {
+				w.WriteHeader(http.StatusServiceUnavailable)
```

Comment:
Just confirming, doing nothing will automatically set `w.WriteHeader(http.StatusOK)` for us, right? I know that happens when `w.Write` is called without calling `w.WriteHeader`. Perhaps we should just make the explicit call for safety/readability?

Reply:
I am okay with being explicit. It won't make any difference to me.